gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ObservationCommentMapperTest.java

```diff
@@ -0,0 +1,141 @@
+        assertThat(identifier.getValue()).isEqualTo(narrativeStatementId);
+        assertThat(identifier.getSystem()).isEqualTo(IDENTIFIER_SYSTEM);
+
+        assertThat(observation.getPerformer().get(0).getReference()).isEqualTo("Practitioner/2D70F602-6BB1-47E0-B2EC-39912A59787D");
+        assertThat(observation.getComment()).isEqualTo("Some example text");
+        assertThat(observation.getContext().getResource()).isEqualTo(encounter);
+    }
+
+    @Test
+    public void mapObservationsWithFullDataMultipleObservations() {
+        var ehrExtract = unmarshallCodeElement("multiple_narrative_statements.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        assertThat(observations.size()).isEqualTo(EXPECTED_OBSERVATION_COUNT);
+    }
+
+    @Test
+    public void mapObservationsNoObservationToMap() {
+        var ehrExtract = unmarshallCodeElement("no_narrative_statement.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        assertThat(observations).isEmpty();
+    }
+
+    @Test
+    public void mapObservationsCompositionHasNoAuthorTime() {
+        var ehrExtract = unmarshallCodeElement("nullflavour_composition_author_time.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        // Value taken from the extract's availability time
+        assertThat(observations.get(0).getIssuedElement().asStringValue()).isEqualTo("2020-01-01T01:01:01.000+00:00");
+    }
+
+    @Test
+    public void mapObservationWithFullDataNoMappedEncounter() {
+        var ehrExtract = unmarshallCodeElement("single_narrative_statement.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        // Calling `getContext` auto creates a Reference object so asserting the reference is null
+        assertThat(observations.get(0).getContext().getReference()).isEqualTo(null);
+    }
+
+    @Test
+    public void mapObservationWithNoComment() {
+        var ehrExtract = unmarshallCodeElement("whitespace_only_text_field.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        assertThat(observations.get(0).getComment()).isEqualTo(null);
+    }
+
+    private RCMRMT030101UK04NarrativeStatement getNarrativeStatement(RCMRMT030101UK04EhrExtract ehrExtract) {
+        return ehrExtract.getComponent().get(0)
+            .getEhrFolder().getComponent().get(0)
+            .getEhrComposition().getComponent().get(0)
+            .getNarrativeStatement();
+    }
+
+    @SneakyThrows
+    private RCMRMT030101UK04EhrExtract unmarshallCodeElement(String fileName) {
```

Comment:
`unmarshallEhrExtractElement` or `unmarshallEhrExtract` could be more suitable name :D

Reply:
Yes, thanks for spotting that